### PR TITLE
Restore and fix macOS dark text selection cursor

### LINKED ISSUE #94144: Blinking cursor outline when selecting with mouse
- VSCode Version: 1.43.2
- OS Version: macOS Catalina 10.15.3 (19D76)

Steps to Reproduce:

1. Select text with a mouse and pay attention to the cursor: its outline would repeatedly appear and disappear at the rate of your movement. This doesn't happen in any other text editors and looks ugly.

Cursor with outline:
outline

Cursor without outline:
no-outline

If you select fast, the transformation happens fast as well, so most users probably don't notice it:
cursor-opt

Here's TextMate for comparison — buttery smooth:
cursor-tm-opt

Does this issue occur when all extensions are disabled?: Yes

### LINKED ISSUE #96946: Text selection cursor is hard to see on older versions of macOS
- VS Code Version: Code - Insiders 1.45.0-insider (fd5d9a30259bb2bebab3e62dfd8315c2e1b27744, 2020-05-04T09:21:17.770Z)
- OS Version: macOS High Sierra 10.13.6

Steps to Reproduce:

1. Switch to a dark theme
2. Move mouse to the text area

Does this issue occur when all extensions are disabled?: Yes

This is a new issue caused by the removal of the special dark cursor in a915df17f58315bf1245d27701f3f7a818a71564 to fix #94144. The default cursor is visible on macOS Catalina, but older versions of macOS don't have a text selection cursor that's visible on dark backgrounds. The old cursor can be fixed without removing it entirely.

### PR DESCRIPTION
**Pull request by kdrag0n**
This PR fixes #96946 by restoring the old cursor for all macOS versions, because even on macOS Catalina, the old special cursor is easier on the eyes when there's a dark background. The continued use of a special cursor can be seen in other apps as well, such as iTerm 2.

The cursor was originally removed in a915df17f58315bf1245d27701f3f7a818a71564 to fix #94144, but removing it entirely isn't necessary. This PR fixes the inconsistency issues exhibited in the old implementation of the special cursor by adding a shared `mouse-text` class to all editor-related elements that need a text selection cursor in their entire bounding box and applying the special cursor to that class rather than just `.monaco-editor .view-lines`.

The custom dark cursor is now consistent in the following areas:

- Text caret (which was the original issue behind #94144)
- Editor text view
- Diff editor
- GitHub review comments and "Reply" field
- Debug console REPL

### CODE REVIEW

**Review by alexdima — APPROVED** [review]
I moved some things around, to get this working also for the standalone editor. The changes in the editor LGTM.

cc @isidorn for a small change to adopt in the repl
cc @rebornix  for the change to adopt in the comments thread

File: src/vs/workbench/contrib/debug/browser/media/repl.css [hunk 1]
```diff
@@ -59,10 +59,6 @@
 	max-width: 150px;
 }
```
Comment (isidorn) [inline src/vs/workbench/contrib/debug/browser/media/repl.css:61]: This line should not be removed. This controls the looks of the cursor inside the debug console tree and is unrelated to this PR.

**Review by isidorn — CHANGES_REQUESTED** [review]

**Comment by isidorn** [comment]
I have added one comment which I think should be tackled. Otherwise the change in `repl.ts` looks good.
I do not understand why is a change in `textAreaHandler` needed? This `textArea` is not visible.

File: src/vs/workbench/contrib/debug/browser/media/repl.css (hunk 1 above)
Comment (alexdima) [reply src/vs/workbench/contrib/debug/browser/media/repl.css:61]: @isidorn That is the entire point of this PR, to change how the mouse cursor looks like on mac, but it goes even outside of the editor and adopts the custom cursor that the editor had also in the repl and in the comments thread.

**Comment by alexdima** [comment]
The textarea can become visible when doing input composition, like e.g. typing Japanese text.

**Comment by isidorn** [comment]
Good point, thanks.

File: src/vs/workbench/contrib/debug/browser/media/repl.css (hunk 1 above)
Comment (isidorn) [reply src/vs/workbench/contrib/debug/browser/media/repl.css:61]: Aha, I see that the class is set on the repl-tree.
I misunderstood the PR that it changes cursor only in the editor and all the editor instances like the repl input.
So I take my original comment back.

**Review by isidorn — APPROVED** [review]

**Comment by kdrag0n** [comment]
I wasn't actually thinking of IMEs when I added the text cursor class to the textarea — it was because it's one of the elements causing #94144 (the other element being the cursor div). It's not necessarily visible, but there's still a 1 px mouse hover area that it occupies, so the cursor changes when moving over the textarea.

**Comment by alexdima** [comment]
@rebornix I have tried this locally and the changes in the comment thread look good to me, I think it makes sense to have the same `I` cursor everywhere, in the editor, and in the comment body/title/reply button.

I will optimistically merge, since this looks good and people are complaining (e.g. #97550) and I'd like to have a fix out in Insiders.

**Comment by rebornix** [comment]
@alexdima thanks for the ping, it looks good to me!

**Pull request merged**